src/App.tsx

```diff
@@ -1,61 +1,157 @@
 import './App.scss';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import React, { useState } from 'react';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { TodoList } from './components/TodoList';
+import { User } from './types/User';
+import { Todo } from './types/Todo';
+
+function getUser(userId: number): User | null {
+  const foundUser = usersFromServer.find(user => user.id === userId);
+
+  return foundUser || null;
+}
+
+const todos: Todo[] = todosFromServer.map(todo => ({
+  ...todo,
+  user: getUser(todo.userId),
+}));
+
+export const App: React.FC = () => {
+  const [newTodos, setNewTodos] = useState(todos);
+  const [invalidInput, setInvalidInput] = useState(false);
+  const [invalidSelect, setInvalidSelect] = useState(false);
+  const [title, setTitle] = useState('');
+  const [selectedUser, setSelectedUser] = useState('0');
+
+  const handleChange = (event: React.ChangeEvent<HTMLInputElement
+  | HTMLSelectElement>) => {
+    const {
+      name,
+      value,
+    } = event.target;
+
+    switch (name) {
+      case 'titleInput':
+        setTitle(value);
+        setInvalidInput(false);
+        break;
+
+      case 'userSelect':
+        setSelectedUser(value);
+        setInvalidSelect(false);
+        break;
+
+      default:
+        break;
+    }
+  };
+
+  const submitChanges = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    if ((title === '') && (selectedUser === '0')) {
+      setInvalidInput(true);
+      setInvalidSelect(true);
+
+      return;
+    }
+
+    if (title === '') {
+      setInvalidInput(true);
+
+      return;
+    }
+
+    if (selectedUser === '0') {
+      setInvalidSelect(true);
+
+      return;
+    }
+
+    setNewTodos([
+      ...newTodos,
+      {
+        id: (Math.max(...newTodos.map(todo => todo.id)) + 1),
```

Comment:
```suggestion
        id: newTodos.length + 1),
```

Reply:
I agree this variant is better, but in this case I can't use it because in 'todosFromServer' we have had todo with id = 15 and length of todos array is 3